clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -227,6 +236,103 @@ object GarbageCollector {
                                                spark,
                                                APIConfigurations(apiURL, accessKey, secretKey)
                                               ).withColumn("run_id", lit(runID))
-    expiredAddresses.write.partitionBy("run_id").mode(SaveMode.Append).parquet(addressesDFLocation)
+    expiredAddresses.write
+      .partitionBy("run_id")
+      .mode(SaveMode.Append)
+      .parquet(addressesDFLocation) // TODO(Guys): consider changing to overwrite
+  }
+}
+
+object S3BulkDeleter {
+  def repartitionBySize(df: DataFrame, maxSize: Int, column: String): DataFrame = {
+    val nRows = df.count()
+    val nPartitions = math.max(1, math.floor(nRows / maxSize)).toInt
```

Comment:
This computes a strange number of partitions, the special case for 1 is strange.  Consider
```suggestion
    val nPartitions = math.ceil(nRows / maxSize).toInt
```
or even just
```suggestion
    val nPartitions = (nRows + maxSize - 1) / maxSize
```